core-httpclient-impl/src/main/java/com/optimizely/ab/OptimizelyFactory.java

```diff
@@ -244,14 +246,19 @@ public static Optimizely newDefaultInstance(String sdkKey, String fallback) {
      * @param sdkKey   SDK key used to build the ProjectConfigManager.
      * @param fallback Fallback datafile string used by the ProjectConfigManager to be immediately available.
      * @param datafileAccessToken  Token for authenticated datafile access.
+     * @param customHttpClient  Customizable CloseableHttpClient to build OptimizelyHttpClient.
      * @return A new Optimizely instance
      */
-    public static Optimizely newDefaultInstance(String sdkKey, String fallback, String datafileAccessToken) {
+    public static Optimizely newDefaultInstance(String sdkKey, String fallback, String datafileAccessToken, CloseableHttpClient customHttpClient) {
```

Comment:
Can we add this new method in addition to existing (sdkKey, fallback, datafileAccessToken)? This change may break for existing clients.